### PR TITLE
Replicated copyright change to open embedded, and resolved multi-license ToDo

### PR DESCRIPTION
**Pull request by shaneallcroft**
#100 Hooray!!!!!!

### CODE REVIEW

**Comment by shaneallcroft** [comment]
@tfoote @allenh1

File: superflore/generators/bitbake/yocto_recipe.py [hunk 1]
```diff
@@ -136,16 +138,8 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
...
             ret += 'LICENSE = "' + self.license + '"\n'
-            """
-            TODO(allenh1): add this functionality
-            first = True
-            for lic in self.license:
-                if not first:
-                    ret += ' '
-                    first = False
-                ret += lic
+            ret += ' '.join(self.license)
```
Comment (tfoote) [inline superflore/generators/bitbake/yocto_recipe.py]: This prints the licences twice, once in a list format and once joined by strings. It's probably worth removing the above print and joining on ', ' for human readability. @allenh1 I assume this is optimized for human readability so comma separated list is best?

Comment (allenh1) [reply superflore/generators/bitbake/yocto_recipe.py]: @tfoote I'm actually not sure how Open Embedded wants multiple licenses to be handled... I think I'll just pull this change locally and try it out? I'm honestly unsure.

File: tests/ebuild/simple_expected.ebuild [hunk 2]
```diff
@@ -8,7 +8,7 @@ inherit ros-cmake
 
 DESCRIPTION="an ebuild"
 HOMEPAGE="https://www.website.com"
-SRC_URI="https://www.website.com/download/${PN}/archive/${PN}/release/lunar/0.0.0.tar.gz -> ${PN}-lunar-release-${PV}.tar.gz"
+SRC_URI="https://www.website.com/download/stuff.tar.gz -> ${PN}-lunar-release-${PV}.tar.gz"
```
Comment (allenh1) [inline tests/ebuild/simple_expected.ebuild]: This change shouldn't be made -- perhaps this got pulled in during the rebase somehow?

Comment (shaneallcroft) [reply tests/ebuild/simple_expected.ebuild]: I'll revert that now :+1:

**Comment by allenh1** [comment]
@shaneallcroft Ok, there's an odd way that OE wants this handled.

```
LICENSE = "GPLv2 & LGPLv2 & BSD & MIT"
```

Shouldn't be that difficult to adapt to, but lmk if you run into any issues.

File: superflore/generators/bitbake/yocto_recipe.py [hunk 3]
```diff
@@ -136,16 +138,8 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
         elif isinstance(self.license, list):
             self.license = self.license[0].replace(' ', '-')
```
Comment (allenh1) [inline superflore/generators/bitbake/yocto_recipe.py]: @shaneallcroft this hackyness needs to be killed with fire. This just removes the list entirely and replaces it with its first element.

Therefore, this line and the one that immediately follows should be removed.

Comment (shaneallcroft) [reply superflore/generators/bitbake/yocto_recipe.py]: You got it! :+1:

File: superflore/generators/bitbake/yocto_recipe.py [hunk 4]
```diff
@@ -134,18 +136,8 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
...
-            ret += 'LICENSE = "' + self.license + '"\n'
-            """
-            TODO(allenh1): add this functionality
-            first = True
-            for lic in self.license:
-                if not first:
-                    ret += ' '
-                    first = False
-                ret += lic
+            ret += 'LICENSE = "' + ' & '.join(self.license)
```
Comment (allenh1) [inline superflore/generators/bitbake/yocto_recipe.py]: This is very close. Make sure to call the `getlicense` function on each entry of the list! Something like the following would probably do the trick.

```py
' & '.join([getlicense(l) for l in self.license])
```

Comment (shaneallcroft) [reply superflore/generators/bitbake/yocto_recipe.py]: OH yeah, that makes sense, cause the hackyness is gone so every item needs to be taken into account im on it :+1:

File: superflore/generators/bitbake/yocto_recipe.py [hunk 5]
```diff
@@ -134,18 +136,8 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
             self.license = self.license.replace(' ', '-')
             ret += 'LICENSE = "' + self.license + '"\n'
```
Comment (allenh1) [inline superflore/generators/bitbake/yocto_recipe.py]: @shaneallcroft You might update this one as well.

Comment (shaneallcroft) [reply superflore/generators/bitbake/yocto_recipe.py]: Oh SNAP i gotchu

**Comment by shaneallcroft** [comment]
:( wait i messed it up

**Comment by allenh1** [comment]
> :( wait i messed it up

It happens. Looks like you just missed an import for the `getlicense` function.

**Comment by shaneallcroft** [comment]
I forgot the underscore i think just a sec

**Comment by shaneallcroft** [comment]
First try :sunglasses:

File: superflore/generators/bitbake/yocto_recipe.py [hunk 6]
```diff
@@ -131,21 +134,11 @@ def get_recipe_text(self, distributor, license_text, die_msg=None):
         self.get_license_line()
         if isinstance(self.license, str):
             self.license = self.license.split(',')[0]
-            self.license = self.license.replace(' ', '-')
-            ret += 'LICENSE = "' + self.license + '"\n'
+            ret += 'LICENSE = "'
+            ret += ' & '.join([get_license(l) for l in self.license]) + '"\n'
         elif isinstance(self.license, list):
```
Comment (allenh1) [inline superflore/generators/bitbake/yocto_recipe.py]: So I don't think you need to use a join when `self.license` is a string. I think this line could just be

```py
ret += 'LICENSE = "%s"\n' % getlicense(self.license)
```

Comment (shaneallcroft) [reply superflore/generators/bitbake/yocto_recipe.py]: Ahhh, i didnt see that '[0]' on line 136, thought it was just being made into a list there so i treated it the same, my b!~

**Comment by shaneallcroft** [comment]
Awesome!!! Thank you so much! : )

**Comment by allenh1** [comment]
(also woot #100 -- that's pretty cool)

**Pull request merged**